### ISSUE TITLE
Prefer array_merge over Array Union Operator

### ISSUE THREAD

**Issue by Etmolf** [einpraegsam/migration#2]
Hi, first of all thx for your work - this extension helped us alot on a current migration project.

One thing i've stumbled upon:

In the File Classes\Port\Export.php on line 277, instead of merging the existing array of relations, you use the array union operator (+) which seems to do the same thing, but it really doesn't.

```
/**
 * Try to find mm relations that should be added
 *
 * @return void
 * @throws DBALException
 */
protected function extendWithMmRelations()
{
	foreach ((array)$this->configuration['relations'] as $table => $configurations) {
		if (DatabaseUtility::isTableExisting($table)) {
			foreach ($configurations as $configuration) {
				$tableMm = $configuration['table'];
				if (DatabaseUtility::isTableExisting($tableMm)) {
					$queryBuilder = DatabaseUtility::getQueryBuilderForTable($tableMm, true);
					$whereClause = $this->getWhereClauseForExtensionWithMmRelations($configuration, $table);
					if ($whereClause !== '') {
						$rows = (array)$queryBuilder
							->select('*')
							->from($tableMm)
							->where($whereClause)
							->execute()
							->fetchAll();
						if (empty($this->jsonArray['mm'][$tableMm])) {
							$this->jsonArray['mm'][$tableMm] = [];
						}
						$this->jsonArray['mm'][$tableMm] += $rows;
					}
				}
			}
		}
	}
}
```

Depending on the number of records already present in the collection, more or less elements from the new collection get ignored (and therefore are lost in the resulting export file).

The solution is to simply use array_merge in this case, as then the resulting array will get reindexed and all data is included.

```
/**
 * Try to find mm relations that should be added
 *
 * @return void
 * @throws DBALException
 */
protected function extendWithMmRelations()
{
	foreach ((array)$this->configuration['relations'] as $table => $configurations) {
		if (DatabaseUtility::isTableExisting($table)) {
			foreach ($configurations as $configuration) {
				$tableMm = $configuration['table'];
				if (DatabaseUtility::isTableExisting($tableMm)) {
					$queryBuilder = DatabaseUtility::getQueryBuilderForTable($tableMm, true);
					$whereClause = $this->getWhereClauseForExtensionWithMmRelations($configuration, $table);
					if ($whereClause !== '') {
						$rows = (array)$queryBuilder
							->select('*')
							->from($tableMm)
							->where($whereClause)
							->execute()
							->fetchAll();
						if (empty($this->jsonArray['mm'][$tableMm])) {
							$this->jsonArray['mm'][$tableMm] = [];
						}
						$this->jsonArray['mm'][$tableMm] = array_merge($this->jsonArray['mm'][$tableMm], $rows);
					}
				}
			}
		}
	}
}
```

I have no possibility to make a pull request right now, so quick & dirty.

Basically only 1 line changes.

Greetings

**Comment by einpraegsam** [comment]
Thx for your feedback. I will include it before the next release.

**Comment by einpraegsam** [comment]
7.3.1 just released
